sdk/aqueduct/backend/api_client.py

```diff
@@ -669,10 +668,10 @@ def get_artifact_result_data(
         if "data" in parsed_response:
             return_value = deserialize(serialization_type, artifact_type, parsed_response["data"])
 
-        if execution_status != ExecutionStatus.SUCCEEDED:
+        if not execution_state or execution_state.status != ExecutionStatus.SUCCEEDED:
```

Comment:
Same question here about handling of DELETED status. Can we get away with just returning `return_value` in that case (if it's None?)

Reply:
@likawind was this addressed? I'm not sure it's clear to me what happens when you call this on a deleted artifact. It would be great if that was in the function's docstring too

Reply:
oh oh this line just prints a warning and doesn't affects the execution flow

Reply:
oh I missed that lol. Can you note that DELETED -> None in the docstring though.

Reply:
yeah updated the warning message and docstring